playwright/_impl/_artifact.py

```diff
@@ -37,7 +37,7 @@ async def path_after_finished(self) -> Optional[pathlib.Path]:
         return pathlib.Path(await self._channel.send("pathAfterFinished"))
 
     async def save_as(self, path: Union[str, Path]) -> None:
-        stream = cast(Stream, from_channel(await self._channel.send("saveAsStream")))
+        stream: Stream = from_channel(await self._channel.send("saveAsStream"))
```

Comment:
I just tried following locally and mypy was happy...

```py
stream: Stream = self
```

Reply:
Stream inherits from ChannelOwner and from_channel returns ChannelOwner hence it is happy